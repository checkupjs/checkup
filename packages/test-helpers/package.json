{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
