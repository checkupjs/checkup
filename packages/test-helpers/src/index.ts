export { default as Project } from './ember-cli-fixturify-project';
export { stdout } from './stdout';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as Plugin } from './plugin';
