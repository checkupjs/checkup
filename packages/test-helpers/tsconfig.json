{
  "extends": "../../tsconfig-base.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "types": ["chai", "jest", "node"],
    "esModuleInterop": true
  },
  "references": [
    {
      "path": "../core"
    }
  ],
  "include": ["src/**/*"]
}
