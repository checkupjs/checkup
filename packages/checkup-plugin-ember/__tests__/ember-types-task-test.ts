import { EmberProject, stdout } from '@checkup/test-helpers';

import EmberTypesTask from '../src/tasks/ember-types-task';
import EmberTypesTaskResult from '../src/results/ember-types-task-result';

const TYPES = {
  components: {
    'my-component.js': '',
  },
  controllers: {
    'my-controller.js': '',
  },
  helpers: {
    'my-helper.js': '',
  },
  initializers: {
    'my-initializer.js': '',
  },
  'instance-initializers': {
    'my-helper.js': '',
  },
  mixins: {
    'my-mixin.js': '',
  },
  models: {
    'my-model.js': '',
  },
  routes: {
    'my-route.js': '',
  },
  services: {
    'my-service.js': '',
  },
  templates: {
    'my-component.hbs': '',
  },
};

describe('types-task', () => {
  let project: EmberProject;

  beforeEach(function () {
    project = new EmberProject('checkup-app', '0.0.0');
  });

  afterEach(function () {
    project.dispose();
  });

  it('returns all the types found in the app and outputs to the console', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: TYPES,
    });

    project.writeSync();

    const result = await new EmberTypesTask({ path: project.baseDir }).run();
    const typesTaskResult = <EmberTypesTaskResult>result;

    typesTaskResult.stdout();

    expect(stdout()).toMatchSnapshot();
  });

  it('returns all the types (including nested) found in the app and outputs to the console', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: TYPES,
    });

    project.addInRepoAddon('ember-super-button', 'latest');

    // @ts-ignore
    project.files.lib['ember-super-button'].addon = TYPES;
    // @ts-ignore

    project.writeSync();

    const result = await new EmberTypesTask({ path: project.baseDir }).run();
    const typesTaskResult = <EmberTypesTaskResult>result;

    typesTaskResult.stdout();

    expect(stdout()).toMatchSnapshot();
  });

  it('returns all the types found in the app and outputs to JSON', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: TYPES,
    });

    project.writeSync();

    const result = await new EmberTypesTask({ path: project.baseDir }).run();
    const typesTaskResult = <EmberTypesTaskResult>result;

    expect(typesTaskResult.json()).toMatchSnapshot();
  });

  it('returns all the types (including nested) found in the app and outputs to JSON', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: TYPES,
    });

    project.addInRepoAddon('ember-super-button', 'latest');

    // @ts-ignore
    project.files.lib['ember-super-button'].addon = TYPES;
    // @ts-ignore

    project.writeSync();

    const result = await new EmberTypesTask({ path: project.baseDir }).run();
    const typesTaskResult = <EmberTypesTaskResult>result;

    expect(typesTaskResult.json()).toMatchSnapshot();
  });

  it('returns all the types found in the app and outputs to html', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: TYPES,
    });

    project.writeSync();

    const result = await new EmberTypesTask({ path: project.baseDir }).run();
    const typesTaskResult = <EmberTypesTaskResult>result;

    expect(typesTaskResult.html()).toMatchSnapshot();
  });

  it('returns all the types (including nested) found in the app and outputs to html', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: TYPES,
    });

    project.addInRepoAddon('ember-super-button', 'latest');

    // @ts-ignore
    project.files.lib['ember-super-button'].addon = TYPES;
    // @ts-ignore

    project.writeSync();

    const result = await new EmberTypesTask({ path: project.baseDir }).run();
    const typesTaskResult = <EmberTypesTaskResult>result;

    expect(typesTaskResult.html()).toMatchSnapshot();
  });
});
