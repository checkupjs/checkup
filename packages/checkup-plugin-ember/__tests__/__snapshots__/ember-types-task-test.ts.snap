// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`types-task returns all the types (including nested) found in the app and outputs to JSON 1`] = `
Object {
  "meta": Object {
    "friendlyTaskName": "Ember Types",
    "taskClassification": Object {
      "category": "insights",
      "priority": "high",
    },
    "taskName": "ember-types",
  },
  "result": Object {
    "types": Array [
      Object {
        "data": Array [
          "addon/components/my-component.js",
          "lib/ember-super-button/addon/components/my-component.js",
        ],
        "displayName": "Components",
        "total": 2,
        "type": "components",
      },
      Object {
        "data": Array [
          "addon/controllers/my-controller.js",
          "lib/ember-super-button/addon/controllers/my-controller.js",
        ],
        "displayName": "Controllers",
        "total": 2,
        "type": "controllers",
      },
      Object {
        "data": Array [
          "addon/helpers/my-helper.js",
          "lib/ember-super-button/addon/helpers/my-helper.js",
        ],
        "displayName": "Helpers",
        "total": 2,
        "type": "helpers",
      },
      Object {
        "data": Array [
          "addon/initializers/my-initializer.js",
          "lib/ember-super-button/addon/initializers/my-initializer.js",
        ],
        "displayName": "Initializers",
        "total": 2,
        "type": "initializers",
      },
      Object {
        "data": Array [
          "addon/instance-initializers/my-helper.js",
          "lib/ember-super-button/addon/instance-initializers/my-helper.js",
        ],
        "displayName": "Instance Initializers",
        "total": 2,
        "type": "instance-initializers",
      },
      Object {
        "data": Array [
          "addon/mixins/my-mixin.js",
          "lib/ember-super-button/addon/mixins/my-mixin.js",
        ],
        "displayName": "Mixins",
        "total": 2,
        "type": "mixins",
      },
      Object {
        "data": Array [
          "addon/models/my-model.js",
          "lib/ember-super-button/addon/models/my-model.js",
        ],
        "displayName": "Models",
        "total": 2,
        "type": "models",
      },
      Object {
        "data": Array [
          "addon/routes/my-route.js",
          "lib/ember-super-button/addon/routes/my-route.js",
        ],
        "displayName": "Routes",
        "total": 2,
        "type": "routes",
      },
      Object {
        "data": Array [
          "addon/services/my-service.js",
          "lib/ember-super-button/addon/services/my-service.js",
        ],
        "displayName": "Services",
        "total": 2,
        "type": "services",
      },
      Object {
        "data": Array [
          "addon/templates/my-component.hbs",
          "lib/ember-super-button/addon/templates/my-component.hbs",
        ],
        "displayName": "Templates",
        "total": 2,
        "type": "templates",
      },
    ],
  },
}
`;

exports[`types-task returns all the types (including nested) found in the app and outputs to html 1`] = `
Array [
  TableData {
    "componentType": "table",
    "formattedResult": Array [
      Object {
        "result": Object {
          "name": "components",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "controllers",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "helpers",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "initializers",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "instance-initializers",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "mixins",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "models",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "routes",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "services",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "templates",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
    ],
    "meta": Object {
      "friendlyTaskName": "Ember Types",
      "taskClassification": Object {
        "category": "insights",
        "priority": "high",
      },
      "taskName": "ember-types",
    },
    "tableHeaders": Array [
      "name",
      "value",
    ],
  },
]
`;

exports[`types-task returns all the types (including nested) found in the app and outputs to the console 1`] = `
"=== Ember Types ===

Types                 Total 
Components            2     
Controllers           2     
Helpers               2     
Initializers          2     
Instance Initializers 2     
Mixins                2     
Models                2     
Routes                2     
Services              2     
Templates             2     

"
`;

exports[`types-task returns all the types found in the app and outputs to JSON 1`] = `
Object {
  "meta": Object {
    "friendlyTaskName": "Ember Types",
    "taskClassification": Object {
      "category": "insights",
      "priority": "high",
    },
    "taskName": "ember-types",
  },
  "result": Object {
    "types": Array [
      Object {
        "data": Array [
          "addon/components/my-component.js",
        ],
        "displayName": "Components",
        "total": 1,
        "type": "components",
      },
      Object {
        "data": Array [
          "addon/controllers/my-controller.js",
        ],
        "displayName": "Controllers",
        "total": 1,
        "type": "controllers",
      },
      Object {
        "data": Array [
          "addon/helpers/my-helper.js",
        ],
        "displayName": "Helpers",
        "total": 1,
        "type": "helpers",
      },
      Object {
        "data": Array [
          "addon/initializers/my-initializer.js",
        ],
        "displayName": "Initializers",
        "total": 1,
        "type": "initializers",
      },
      Object {
        "data": Array [
          "addon/instance-initializers/my-helper.js",
        ],
        "displayName": "Instance Initializers",
        "total": 1,
        "type": "instance-initializers",
      },
      Object {
        "data": Array [
          "addon/mixins/my-mixin.js",
        ],
        "displayName": "Mixins",
        "total": 1,
        "type": "mixins",
      },
      Object {
        "data": Array [
          "addon/models/my-model.js",
        ],
        "displayName": "Models",
        "total": 1,
        "type": "models",
      },
      Object {
        "data": Array [
          "addon/routes/my-route.js",
        ],
        "displayName": "Routes",
        "total": 1,
        "type": "routes",
      },
      Object {
        "data": Array [
          "addon/services/my-service.js",
        ],
        "displayName": "Services",
        "total": 1,
        "type": "services",
      },
      Object {
        "data": Array [
          "addon/templates/my-component.hbs",
        ],
        "displayName": "Templates",
        "total": 1,
        "type": "templates",
      },
    ],
  },
}
`;

exports[`types-task returns all the types found in the app and outputs to html 1`] = `
Array [
  TableData {
    "componentType": "table",
    "formattedResult": Array [
      Object {
        "result": Object {
          "name": "components",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "controllers",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "helpers",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "initializers",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "instance-initializers",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "mixins",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "models",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "routes",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "services",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "templates",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
    ],
    "meta": Object {
      "friendlyTaskName": "Ember Types",
      "taskClassification": Object {
        "category": "insights",
        "priority": "high",
      },
      "taskName": "ember-types",
    },
    "tableHeaders": Array [
      "name",
      "value",
    ],
  },
]
`;

exports[`types-task returns all the types found in the app and outputs to the console 1`] = `
"=== Ember Types ===

Types                 Total 
Components            1     
Controllers           1     
Helpers               1     
Initializers          1     
Instance Initializers 1     
Mixins                1     
Models                1     
Routes                1     
Services              1     
Templates             1     

"
`;
